src/Utils/SortedArray.vala

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2019-2021 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Ashish Shevale <shevaleashish@gmail.com>
+ */
+
+ /*
+  * This data structure will be used for storing the positions of guidelines.
+  * Storing all elements in a sorted array will make it easier to check if a guideline
+  * exists at the given position, to get the nearest neighbourest neightbours.
+  * Since this data structure does not allow storing duplicate values,
+  * placing a guideline over another will automatically delete it.
+  *
+  */
+ public class Akira.Utils.SortedArray : Object {
```

Comment:
I have several issues with this.
1. This is not really a sorted array as such. It's a set with bounds.
2. It's very specifically programmed for guidelines, which makes the generic looking name misleading
3. It's not very efficient, even for binary search (this could be easily fixed)
4. It probably makes more sense to use a Gee.TreeSet in a wrapper

Reply:
I wrapped a TreeSet in the SortedArray. Also renamed SortedArray to SortedGuides. 

> Mainly the issue I have is storing some guidelinemodels in the instance and other in the manager

I don't understand what this means. The GuidelineManager only stores a reference to the models from the instance. This way, based on the events, the guide data for the artboard instance is directly changed.